### ISSUE TITLE
Eslint Migration - Nextjs

### ISSUE THREAD

**Issue by mouadbelbah-planoly** [biomejs/biome#2935]
### Environment information

```block
CLI:
  Version:                      1.7.3
  Color support:                true

Platform:
  CPU Architecture:             aarch64
  OS:                           macos

Environment:
  BIOME_LOG_DIR:                unset
  NO_COLOR:                     unset
  TERM:                         "xterm-256color"
  JS_RUNTIME_VERSION:           "v20.13.1"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         "npm/10.5.2"

Biome Configuration:
  Status:                       Loaded successfully
  Formatter disabled:           false
  Linter disabled:              false
  Organize imports disabled:    false
  VCS disabled:                 false

Workspace:
  Open Documents:               0
```

### What happened?

1. Started a new Nextjs Project with Eslint
2. Installed biome following the get-started guide.
3. ran `npx biome migrate eslint --write` 

Received Error:

 Migration has encountered an error: The module '..../node_modules/eslint-config-next/index.js' cannot be loaded. Make sure that the module exists.

### Expected result

It should migrate Nextjs Eslint config

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by ematipico** [comment]
Please provide us with a minimal reproduction

**Comment by Conaclos** [comment]
Some idea: If you use a package manager that doesn't make accessible indirect dependencies, then Biome cam not be able to load these dependencies. Anyway we need more info to understand your issue.

**Comment by chribjel** [comment]
the repro is very simple:

```sh
npm create next-app # Yes to everthing
npm install --save-dev --save-exact @biomejs/biome
npx @biomejs/biome init
npx @biomejs/biome migrate eslint --write
```

**Comment by Conaclos** [comment]
The reported error is not so informative and even misleading. I will open a PR to fix that.

With a proper error propagation I am getting the following error:

```
  ✖ Migration has encountered an error: `node` was invoked to resolve '/playroom/my-app/node_modules/eslint-config-next/index.js'. This invocation failed with the following error:
    /playroom/my-app/node_modules/@rushstack/eslint-patch/lib/_patch-base.js:167
                throw new Error('Failed to patch ESLint because the calling module was not recognized.\n' +
                      ^
    
    Error: Failed to patch ESLint because the calling module was not recognized.
    If you are using a newer ESLint version that may be unsupported, please create a GitHub issue:
    https://github.com/microsoft/rushstack/issues
        at Object.<anonymous> (/playroom/my-app/node_modules/@rushstack/eslint-patch/lib/_patch-base.js:167:19)
        at Module._compile (node:internal/modules/cjs/loader:1455:14)
        at Module._extensions..js (node:internal/modules/cjs/loader:1534:10)
        at Module.load (node:internal/modules/cjs/loader:1265:32)
        at Module._load (node:internal/modules/cjs/loader:1081:12)
        at Module.require (node:internal/modules/cjs/loader:1290:19)
        at require (node:internal/modules/helpers:188:18)
        at Object.<anonymous> (/playroom/my-app/node_modules/@rushstack/eslint-patch/lib/modern-module-resolution.js:11:23)
        at Module._compile (node:internal/modules/cjs/loader:1455:14)
        at Module._extensions..js (node:internal/modules/cjs/loader:1534:10)
    
    Node.js v22.0.0
```

Under the hood Biome executes the ESLint configuration files.
The problem may come from using different ESLint version in the dependency tree.
This is clearly not a Biome issue.

**Comment by chribjel** [comment]
Even though it is not a Biome issue, it is an issue with the migration tool, no? Maybe I am misunderstanding.

The ESLint config that comes with Next works, thus the migration tool, if perfect, should be able to migrate it(?)

I just want what is best for adoption of Biome. Next is a really popular framework and i believe that if the migration tool works for Next, the adoption would increase a lot.

**Comment by ematipico** [comment]
Our migration tool is a **best effort** tool. 

While we try to do our best to accommodate possible bugs on our end, bugs that could come from other plugins can't be solved on our end.

**Comment by marmotz** [comment]
Ok, and how can we migrate manually ?

**Comment by Conaclos** [comment]
The issue seems to be the same as https://github.com/microsoft/rushstack/issues/4301

Biome uses node to run the file. If you execute the following command, you will get the same error.

```shell
node ./node_modules/eslint-config-next/index.js
```

I think that ESLint encounters the issue, but silence this?

> Ok, and how can we migrate manually ?

Reading the stack trace, you could just remove the line 56in `./node_modules/eslint-config-next/index.js` and executes `biome migrate sslint`. This results in the following config:

```
{
	"linter": {
		"enabled": true,
		"rules": {
			"recommended": false,
			"a11y": {
				"noAriaUnsupportedElements": "warn",
				"noBlankTarget": "off",
				"useAltText": "warn",
				"useAriaPropsForRole": "warn",
				"useValidAriaProps": "warn",
				"useValidAriaValues": "warn"
			},
			"correctness": {
				"noChildrenProp": "error",
				"useExhaustiveDependencies": "warn",
				"useHookAtTopLevel": "error",
				"useJsxKeyInIterable": "error"
			},
			"security": { "noDangerouslySetInnerHtmlWithChildren": "error" },
			"suspicious": { "noCommentText": "error", "noDuplicateJsxProps": "error" }
		}
	}
}
```

**Comment by anandabhi04** [comment]
> The reported error is not so informative and even misleading. I will open a PR to fix that.
> 
> With a proper error propagation I am getting the following error:
> 
> ```
>   ✖ Migration has encountered an error: `node` was invoked to resolve '/playroom/my-app/node_modules/eslint-config-next/index.js'. This invocation failed with the following error:
>     /playroom/my-app/node_modules/@rushstack/eslint-patch/lib/_patch-base.js:167
>                 throw new Error('Failed to patch ESLint because the calling module was not recognized.\n' +
>                       ^
>     
>     Error: Failed to patch ESLint because the calling module was not recognized.
>     If you are using a newer ESLint version that may be unsupported, please create a GitHub issue:
>     https://github.com/microsoft/rushstack/issues
>         at Object.<anonymous> (/playroom/my-app/node_modules/@rushstack/eslint-patch/lib/_patch-base.js:167:19)
>         at Module._compile (node:internal/modules/cjs/loader:1455:14)
>         at Module._extensions..js (node:internal/modules/cjs/loader:1534:10)
>         at Module.load (node:internal/modules/cjs/loader:1265:32)
>         at Module._load (node:internal/modules/cjs/loader:1081:12)
>         at Module.require (node:internal/modules/cjs/loader:1290:19)
>         at require (node:internal/modules/helpers:188:18)
>         at Object.<anonymous> (/playroom/my-app/node_modules/@rushstack/eslint-patch/lib/modern-module-resolution.js:11:23)
>         at Module._compile (node:internal/modules/cjs/loader:1455:14)
>         at Module._extensions..js (node:internal/modules/cjs/loader:1534:10)
>     
>     Node.js v22.0.0
> ```
> 
> Under the hood Biome executes the ESLint configuration files. The problem may come from using different ESLint version in the dependency tree. This is clearly not a Biome issue.

I am having the same issue how can i fix it.

**Comment by anandabhi04** [comment]
I don't have ./node_modules/eslint-config-next/index.js in my repo

**Comment by Conaclos** [comment]
> I am having the same issue how can i fix it.
> I don't have ./node_modules/eslint-config-next/index.js in my repo

Could you open a new issue with detailed information? It is hard to help without any information.

**Comment by aksh1618** [comment]
> Reading the stack trace, you could just remove the line that imports `@rushstack/eslint-patch/modern-module-resolution` in `./node_modules/eslint-config-next/index.js` and executes `biome migrate sslint`. This results in the following config:

It might be worthwhile adding this as a note in the migration doc, as this fix is likely applicable for any given NextJS project.

**Comment by estevan-ulian** [comment]
> The issue seems to be the same as [microsoft/rushstack#4301](https://github.com/microsoft/rushstack/issues/4301)
> 
> Biome uses node to run the file. If you execute the following command, you will get the same error.
> 
> ```shell
> node ./node_modules/eslint-config-next/index.js
> ```
> 
> I think that ESLint encounters the issue, but silence this?
> 
> > Ok, and how can we migrate manually ?
> 
> Reading the stack trace, you could just remove the line that imports `@rushstack/eslint-patch/modern-module-resolution` in `./node_modules/eslint-config-next/index.js` and executes `biome migrate sslint`. This results in the following config:
> 
> ```json
> {
> 	"linter": {
> 		"enabled": true,
> 		"rules": {
> 			"recommended": false,
> 			"a11y": {
> 				"noAriaUnsupportedElements": "warn",
> 				"noBlankTarget": "off",
> 				"useAltText": "warn",
> 				"useAriaPropsForRole": "warn",
> 				"useValidAriaProps": "warn",
> 				"useValidAriaValues": "warn"
> 			},
> 			"correctness": {
> 				"noChildrenProp": "error",
> 				"useExhaustiveDependencies": "warn",
> 				"useHookAtTopLevel": "error",
> 				"useJsxKeyInIterable": "error"
> 			},
> 			"security": { "noDangerouslySetInnerHtmlWithChildren": "error" },
> 			"suspicious": { "noCommentText": "error", "noDuplicateJsxProps": "error" }
> 		}
> 	}
> }
> ```

These steps work for me, but using the following command:

```
npx @biomejs/biome migrate eslint --write
```

I'm using next 15.0.1

**Comment by maronnjapan** [comment]
Thank you for checking and addressing the issue ([#4528](https://github.com/biomejs/biome/issues/4528)) I posted  last week.
I wanted to test the fixes, so I compiled the Rust files from the main branch and integrated them into my project, then ran the migration command.
However, the same error occurred and I was unable to migrate.
If I want to test the functionality, should I wait for a release that includes the fixed code rather than replacing it with compiled files?

**Comment by Conaclos** [comment]
@maronnjapan I just released a [new nightly version of Biome](https://github.com/biomejs/biome/releases/tag/cli%2Fv1.9.5-nightly.c0cccb2). You can use it for testing the fixes :)

**Comment by maronnjapan** [comment]
@Conaclos 
Thank you for releasing the nightly version.
I tried installing it and running the migration command, and the "ERR_MODULE_NOT_FOUND" error no longer occurs!
Thank you very much !!!!

**Comment by heecheon92** [comment]
> The issue seems to be the same as [microsoft/rushstack#4301](https://github.com/microsoft/rushstack/issues/4301)
> 
> Biome uses node to run the file. If you execute the following command, you will get the same error.
> 
> node ./node_modules/eslint-config-next/index.js
> I think that ESLint encounters the issue, but silence this?
> 
> > Ok, and how can we migrate manually ?
> 
> Reading the stack trace, you could just remove the line that imports `@rushstack/eslint-patch/modern-module-resolution` in `./node_modules/eslint-config-next/index.js` and executes `biome migrate sslint`. This results in the following config:
> 
> {
> 	"linter": {
> 		"enabled": true,
> 		"rules": {
> 			"recommended": false,
> 			"a11y": {
> 				"noAriaUnsupportedElements": "warn",
> 				"noBlankTarget": "off",
> 				"useAltText": "warn",
> 				"useAriaPropsForRole": "warn",
> 				"useValidAriaProps": "warn",
> 				"useValidAriaValues": "warn"
> 			},
> 			"correctness": {
> 				"noChildrenProp": "error",
> 				"useExhaustiveDependencies": "warn",
> 				"useHookAtTopLevel": "error",
> 				"useJsxKeyInIterable": "error"
> 			},
> 			"security": { "noDangerouslySetInnerHtmlWithChildren": "error" },
> 			"suspicious": { "noCommentText": "error", "noDuplicateJsxProps": "error" }
> 		}
> 	}
> }

This fixed the issue for the eslint migration. thanks

**Comment by codefug** [comment]
I found another possible source of this error.

In my case, using eslint-plugin-storybook caused the same issue during migration. The migration failed with the following error:

> Migration has encountered an error: node was invoked to resolve './eslint.config.mjs' …

As a workaround, I backed up the original eslint.config.mjs, temporarily removed the configuration related to the Storybook plugin, and then ran the migration again. After the migration completed successfully, I restored the Storybook-related settings.

I’m sharing this in case it helps others who are stuck on the same issue like I was.

**Comment by dyc3** [comment]
@codefug This issue is very stale. Your issue is likely different from this one. If you could file a new issue with a minimal repro, that would be super helpful.

**Comment by codefug** [comment]
> [@codefug](https://github.com/codefug) This issue is very stale. Your issue is likely different from this one. If you could file a new issue with a minimal repro, that would be super helpful.

Thanks! I’ve checked it. I’ll create a new issue!
